### PR TITLE
fix: update default query of error page

### PR DESCRIPTION
**Pull request by tuin77**
## Description
fixed https://github.com/vbenjs/vue-vben-admin/issues/4617
组件：vxe-table封装
问题：添加筛选条件并点击翻页，会导致页面查询结果可能不正确(页面返回了第一页，但没有数据)
解决：根据查询返回的总数量判断当前页码是否正确，不正确则重新请求
问题复现：先把筛选条件清空查询，复制从第一页列表已有的值输入Category输入框中，点击页码3查询结果不应该为空；

## Type of change

Please delete options that are not relevant.

- [ ] Bug fix (non-breaking change which fixes an issue)
- [ ] New feature (non-breaking change which adds functionality)
- [ ] Breaking change (fix or feature that would cause existing functionality to not work as expected)
- [ ] This change requires a documentation update
- [ ] Please, don't make changes to `pnpm-lock.yaml` unless you introduce a new test example.

## Checklist

> ℹ️ Check all checkboxes - this will indicate that you have done everything in accordance with the rules in [CONTRIBUTING](contributing.md).

- [ ] If you introduce new functionality, document it. You can run documentation with `pnpm run docs:dev` command.
- [ ] Run the tests with `pnpm test`.
- [ ] Changes in changelog are generated from PR name. Please, make sure that it explains your changes in an understandable manner. Please, prefix changeset messages with `feat:`, `fix:`, `perf:`, `docs:`, or `chore:`.
- [ ] My code follows the style guidelines of this project
- [ ] I have performed a self-review of my own code
- [ ] I have commented my code, particularly in hard-to-understand areas
- [ ] I have made corresponding changes to the documentation
- [ ] My changes generate no new warnings
- [ ] I have added tests that prove my fix is effective or that my feature works
- [ ] New and existing unit tests pass locally with my changes
- [ ] Any dependent changes have been merged and published in downstream modules

## Summary by CodeRabbit

- **New Features**
	- Introduced a filtering capability for mock data, allowing users to retrieve specific subsets based on defined criteria.

- **Improvements**
	- Enhanced pagination handling to ensure the current page adjusts correctly when the total number of items changes, improving user experience during data navigation.

### CODE REVIEW

**Comment by anncwb** [comment]
ai有些建议。
然后我的想法是，输入框的参数，在没点击查询之前，应该是不能被应用的，所以可能点击跳转页数，可能不需要携带这个新的参数，应该是需要旧的参数。这样分页应该也不会有问题。

**Comment by tuin77** [comment]
沿用v2版本的逻辑，感觉也更符合用户的预期，用户输入筛选值点击翻页自然是想查询，俺家产品倾向这种结果😄

File: packages/effects/plugins/src/vxe-table/extends.ts [hunk 1]
```diff
@@ -38,6 +38,18 @@ function extendProxyOption(
   const wrapperFn = async (params: any, _formValues: any, ...args: any[]) => {
     const formValues = getFormValues();
     const data = await configFn(params, formValues, ...args);
+
+    // If the page is incorrect, make corrections and request again
+    const total = proxyConfig?.response?.total as string;
```
[the remark this answers is not in the document]
Comment (tuin77) [reply packages/effects/plugins/src/vxe-table/extends.ts:43]: 这里的total类型不知该怎么处理😭

**Comment by ZhangXY-Magic** [comment]
> 沿用v2版本的逻辑，感觉也更符合用户的预期，用户输入筛选值点击翻页自然是想查询，俺家产品倾向这种结果😄

我提这个问题也是倾向于未点击查询，筛选参数不应生效，我看了下antd的protable也是这种逻辑

File: packages/effects/plugins/src/vxe-table/extends.ts [hunk 2]
```diff
@@ -38,6 +38,18 @@ function extendProxyOption(
...
+    const { pageSize, currentPage } = params.page;
+    const resultTotal: number = data[total];
```
[the remark this answers is not in the document]
Comment (tuin77) [reply packages/effects/plugins/src/vxe-table/extends.ts]: 已添加对嵌套result.total类型支持

**Comment by tuin77** [comment]
> > 沿用v2版本的逻辑，感觉也更符合用户的预期，用户输入筛选值点击翻页自然是想查询，俺家产品倾向这种结果😄
> 
> 我提这个问题也是倾向于未点击查询，筛选参数不应生效，我看了下antd的protable也是这种逻辑

两种都支持 做成配置项？

**Comment by anncwb** [comment]
这个有时间我再看下，先放着

**Pull request closed without merging**